src/Extensions/ShareDraftContentSiteTreeExtension.php

```diff
@@ -55,8 +56,14 @@ public function ShareTokenLink()
     {
         $shareToken = $this->getNewShareToken();
 
+        if (class_exists(Subsite::class) && $this->owner->SubsiteID) {
+            $link = $this->owner->Subsite()->domain();
+        } else {
+            $link = Director::absoluteBaseURL();
```

Comment:
I feel like a way of avoiding this coupling would be for subsites to change the result of this function. This is fine for now though, thanks. We'll merge and investigate further improvements separately.